### PR TITLE
refactor: reduce code complexity

### PR DESCRIPTION
**Pull request by SimonSchick**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

- Cleanups remaining `else (if)`'s with linter help.
- Fixed various lint warnings and made them errors so they don't come back
- Removed some dead node 0.10.x support snippet
- Replace `.npmignore` with `files` in `package.json` and exclude all redundant files for use.

On another note, is there a reason you call `Error.captureStackTrace(this, this.constructor);` on all error errors?
Are you doing to to prevent it from capturing the first stack frame?

This will be my last code cleanup PR until v5 is out.

### CODE REVIEW

**Comment by SimonSchick** [comment]
Hey @sushantdhiman you think we can get my last 2 PRs in before v5 goes live? :)

File: package.json [hunk 1]
```diff
@@ -21,6 +21,9 @@
   "engines": {
     "node": ">=6.0.0"
   },
+  "files": [
+    "lib"
```
Comment (sushantdhiman) [inline package.json]: Need to include `index.js`

Comment (SimonSchick) [reply package.json]: Nope, the `mail` file is automatically added :)

Comment (SimonSchick) [reply package.json]:
```
$ npm pack
npm notice
npm notice package: sequelize@5.0.0-beta.14
npm notice === Tarball Contents ===
npm notice 6.2kB   package.json
npm notice 116B    index.js
...
```

**Comment by sushantdhiman** [comment]
Thanks @SimonSchick

**Pull request merged**